zeroconf/__init__.py

```diff
@@ -2617,6 +2617,22 @@ def can_send_to(sock: socket.socket, address: str) -> bool:
     return cast(bool, addr.version == 6 if sock.family == socket.AF_INET6 else addr.version == 4)
 
 
+def autodetect_ip_version(interfaces: InterfacesType) -> IPVersion:
+    """Auto detect the IP version when it is not provided."""
+    if isinstance(interfaces, list):
```

Comment:
Should reverse the condition and outdent in a followup, but I didn't want to alter the comparability in the same PR